src/mo_types/type.ml

```diff
@@ -64,15 +79,15 @@ and scope = typ
 and bind_sort = Scope | Type
 
 and bind = {var : var; sort: bind_sort; bound : typ}
-and src = {depr : string option; region : Source.region}
+and src = {depr : string option; region : Source.region; mutable srcs : Region_set.t}
```

Comment:
Yikes! do we really need to make the type mutable for this?

Reply:
This seemed like the least intrusive way to propagate occurrences between fields when we are running `eq` or `sub`. Having a very complicated solution with a bigger maintenance burden for the language server seemed suboptimal.